DashAISoftware/DashAI
# Modify ItemsList.jsx to make it more reusable

# Summary


Change `ItemsList.jsx` to `ItemSelector.jsx` which is a more general version for reuse in other components, these modifications also allow to keep the values selected by the user in case he goes back to previous steps in the modal.

## Type of change


- Front end new feature.
- Refactoring

## Changes


- Created the `ItemSelector.jsx` component as a general version of a list of selectable items that displays a description of the selected item.
- Created `SelectTaskStep.jsx` and `SelectDataloaderStep.jsx` as steps of the dataset modal that use `ItemSelector.jsx`
- Added as a new feature the ability to keep the user's selection even if the user changes the step.

## How to Test


1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()`
3. Navigate to the datasets tab and create a new dataset by clicking the "New Dataset" button
4.  Select one of the tasks
5. Select one of the dataloaders
6. Navigate back to the step where a task is selected and you will see that the task you selected earlier is still selected.

## Screenshots


![itemSelector (1)](https://github.com/DashAISoftware/DashAI/assets/53657198/599cdb02-d227-45dd-88cb-21fefed3bd41)


## Notes


This PR depends on #73
